samples/gibbs_ensemble/gibbs.py

```diff
@@ -0,0 +1,139 @@
+import socket
+import struct
+import pickle
+import numpy as np
+
+
+class MessageId:
+    START = 0
+    END = 1
+    MOVE_PART = 2
+    MOVE_PART_REVERT = 21
+    CHANGE_VOLUME = 3
+    CHANGE_VOLUME_REVERT = 31
+    EXCHANGE_PART_ADD = 4
+    EXCHANGE_PART_ADD_REVERT = 41
+    EXCHANGE_PART_REMOVE = 5
+    EXCHANGE_PART_REMOVE_REVERT = 51
+    ENERGY = 6
+
+
+class Client():
+    '''
+    Base class for a Gibbs ensemble client. Each client runs a single
+    system, executes requested changes and returns energies.
+    '''
+    _system = None
+
+    def __init__(self, system):
+        self._system = system
+
+    def random_particle(self):
+        id = np.random.choice(self._system.part[:].id)
+        return self._system.part[id]
+
+    def move_particle(self):
+        '''Moves a particle to a new random position. The old position is saved in _old_part_position.'''
+        p = self.random_particle()
+        self._last_modified_pid = p.id
+        self._saved_pos = p.pos
```

Comment:
don't you have to copy here?

Reply:
No. Instances of ArrayLocked contain a copy of the data. They are no longer tied to the underlying particle. Only write access to elements are blocked. This variable is only assigned and read.